player.gd

```diff
@@ -20,33 +23,36 @@ func _ready():
 func _process(delta):
 	var direction = 0
 	
+
+	if in_air:
+		speed -= air_gravity * delta
 	# Get direction player is swimming, but require a keystoke per impulse
-	if Input.is_action_just_pressed("swim_up"):
-		direction = 1
-	if Input.is_action_just_pressed("swim_down"):
-		direction = -1
-	
-	# Apply the swim impulse
-	speed += swim_impulse * direction
-	
-	# Apply constant drag against movement
-	if speed > coasting_deceleration * delta: 
-		speed -= coasting_deceleration * delta
-	elif speed < -(coasting_deceleration * delta):
-		speed += coasting_deceleration * delta
 	else:
-		speed = 0
+		if Input.is_action_just_pressed("swim_up"):
+			direction = 1
+		if Input.is_action_just_pressed("swim_down"):
+			direction = -1
+		# Apply the swim impulse
+		speed += swim_impulse * direction
```

Comment:
I hadn't intended to imitate the "flap" part of flappy bird -- I was imagining building up speed and popping out of the water being the main challenge, so it might be nice to be able to just hold down the arrow keys

Reply:
This could always be changed if it makes sense. Maybe we could try out each and see how they compare. I guess now that I think about it, fish do accelerate at a constant rate, generally.